### ISSUE TITLE
Discussion of Tutorial 2.4

### ISSUE THREAD

**Issue by stevevanhooser** [VH-Lab/NDI-matlab#179]
Enter your comments, questions, bugs, or discussion items for Tutorial 2.4 here.

**Comment by KellyZhangL** [comment]
When I run Codeblock 2.4.4.1 `e = S.getelements('element.type','spikes');`, e returns a _0x0 cell_. I tried clearing clearing (`S.databas_clear('yes')`) and reopening the session (`ndi.setup.vhlab`) but e returned the same value.

**Comment by stevevanhooser** [comment]
Hi Kelly - Tutorial 2.3 has to be done before 2.4 will work. The results depend on it. I added a note to that effect in Tutorial 2.4 now.

**Comment by KellyZhangL** [comment]
I'm not sure if this is an error but when I run Code block 2.4.5.1, many graphs and (it seems) duplicates of these graphs pop up. The first time I ran Code block 2.4.5.1, I got 6 graphs. This time, I got 15 graphs (1, 2, 3 seem to be duplicates; 4, 5, 6 seem to be duplicates; 7, 8, 9 seem to be duplicates; 10, 11; 12, 13; and 14, 15).

**Comment by KellyZhangL** [comment]
Code block 2.4.5.2: I believe the following are missing an additional {1} ( `oriprops{1}.document` returns a dot indexing error while `oriprops{1}{1}.document` works):

`oriprops{1}.document_properties.orientation_direction_tuning.significance`

`oriprops{1}.document_properties.orientation_direction_tuning.fit`

`oriprops{1}.document_properties.orientation_direction_tuning.vector`

**Comment by stevevanhooser** [comment]
> I'm not sure if this is an error but when I run Code block 2.4.5.1, many graphs and (it seems) duplicates of these graphs pop up. The first time I ran Code block 2.4.5.1, I got 6 graphs. This time, I got 15 graphs (1, 2, 3 seem to be duplicates; 4, 5, 6 seem to be duplicates; 7, 8, 9 seem to be duplicates; 10, 11; 12, 13; and 14, 15).

Look closely..one is probably the F0 response, another is the F1 response, and another is the F2 response. The shapes should be a little different. The F0 should be the largest for ctx_1 and F1 for lgn_1.  Maybe I should include some photos of what is expected with some commentary?

**Comment by stevevanhooser** [comment]
> Code block 2.4.5.2: I believe the following are missing an additional {1} ( `oriprops{1}.document` returns a dot indexing error while `oriprops{1}{1}.document` works):
> 
> `oriprops{1}.document_properties.orientation_direction_tuning.significance`
> 
> `oriprops{1}.document_properties.orientation_direction_tuning.fit`
> 
> `oriprops{1}.document_properties.orientation_direction_tuning.vector`

Thanks, this is fixed! Good find

**Comment by stevevanhooser** [comment]
> > I'm not sure if this is an error but when I run Code block 2.4.5.1, many graphs and (it seems) duplicates of these graphs pop up. The first time I ran Code block 2.4.5.1, I got 6 graphs. This time, I got 15 graphs (1, 2, 3 seem to be duplicates; 4, 5, 6 seem to be duplicates; 7, 8, 9 seem to be duplicates; 10, 11; 12, 13; and 14, 15).
> 
> Look closely..one is probably the F0 response, another is the F1 response, and another is the F2 response. The shapes should be a little different. The F0 should be the largest for ctx_1 and F1 for lgn_1. Maybe I should include some photos of what is expected with some commentary?

I added a picture in Tutorial 2.4. See how you like it!

**Comment by KellyZhangL** [comment]
Thank you! These images are very helpful for comparison. I did have duplicates but I'm going to rerun Tutorial 2.4 and see if this happens again.

**Comment by KellyZhangL** [comment]
I reran Codeblock 2.4.5.1 and I got the correct number of graphs and the same 6 graphs as pictured. I think I got duplicates because I got impatient and reran the code while it was still running.

**Comment by KellyZhangL** [comment]
I'm rerunning all the code for profile analyses (I'm just copying and pasting this time so I haven't mistyped anything) and rdocs is a _1x1 cell_ so I'm unable to run `rdocs{2} = rapp.stimulus_responses(stimprobe, e{2}, redo);` properly.

**Comment by KellyZhangL** [comment]
Reran everything again and rdocs is a _1x2 cell_ so everything is working.

**Comment by KellyZhangL** [comment]
Running a profile analysis on code block 2.4.5.1 
`for i=1:2,
    tdoc{i} = oapp.calculate_tuning_curve(e{i});
    oriprops{i} = oapp.calculate_all_oridir_indexes(e{i}); 
end;`

is returning the following error:

> Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 89)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)
 
Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 96)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)
 
Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 89)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)
 
Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 96)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)
 
Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 89)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)
 
Warning: Matrix is singular to working precision. 
> In vlt.stats.hotellingt2test (line 27)
In vlt.neuro.vision.oridir.index.oridir_vectorindexes
(line 96)
In ndi.app/oridirtuning/calculate_oridir_indexes
(line 134)
In ndi.app/oridirtuning/calculate_all_oridir_indexes
(line 72)

The figures appear as attached. 
![Screenshot 2021-06-06 014603](https://user-images.githubusercontent.com/83798182/120914029-52537900-c669-11eb-83a3-04b55c4022d3.png)
